design/KEPS/multicluster/proposal.md

```diff
@@ -136,13 +136,18 @@ This process will need to be repeated for every new cluster added. Adding or rem
 
-By default, the Kiali Operator will autodiscover all remote cluster secrets found in the Kiali deployment namespace and will mount them to the Kiali Server pod at install time. When the Kiali Server starts up, it will obtain all the necessary remote cluster connection information from the mounted secrets.
+### istioctl x create-remote-secret command
+
+The `istioctl` CLI tool has a command `x create-remote-secret` that is used to create remote cluster secrets that Istio itself uses to communicate to the remote clusters. The original thought was that Kiali could simply use these same secrets for its own communications with the remote clusters. However, for the following reasons, this cannot be done - Kiali cannot re-use the remote cluster secrets created by istioctl.
+1. By default, istioctl will generate remote cluster secrets that are associated with the service account "istio-reader-service-account". That Istio service account does not have the permissions that the Kiali server needs. If Kiali tries to use that remote cluster secret to communicate with the remote cluster, it will encounter multiple failures and will likely not even be able to properly initialize.
+2. The istioctl command does have the ability to associate a remote cluster secret with a custom service account, however, Kiali cannot guarantee that users will do this. All the Istio multicluster docs provide instructions that assume the default service account is used. Also, the custom service account will have to have a superset of all the permissions that both Istio and Kiali need, and users may not want to create such a Service Account and give those permissions to Istio.
+3. When the istioctl command creates a remote cluster secret, it puts a label on the secret to identify the secret as a remote cluster secret. That label name is hardcoded (it is not customizable) with a value of `istio/multiCluster=true`. This is the label that the Kiali Operator/Helm Charts will use to autodiscover remote cluster secrets (e.g. `kubectl get secrets -n istio-system -l istio/multiCluster=true`). This will be a problem if the user has multiple secrets where some were created using the "istio-reader-service-account" SA and some with a Kiali-specific SA. For example, assume the user created a remote cluster secret using istioctl with a custom service account that contains the necessary Kiali permissions (so Kiali can communicate with the remote cluster). Also assume the user followed the Istio docs and created a remote cluster secret using the istioctl defaults so Istio can communicate with the remote cluster. There are now two remote cluster secrets, each labeled with `istio/multiCluster=true`. Kiali will auto-discover and use both. But only one will work with Kiali; the other will cause Kiali to encounter multiple failures and will likely cause Kiali to fail to initialize properly.
```

Comment:
I guess it's very typical that Kiali will be deployed alongside Istiod, and I think users will use istioctl cli to create the remote cluster secrets and then, they will create the kiali's specific remote cluster secrets, so I think we might need to change the label that Kiali use to pick the secrets up?.

Reply:
I do mention this on the next line below.. see line 150:

`The idea is that the Kiali tool will generate its own secret with its own identification label (`kiali.io/multiCluster=true`)`

Reply:
Oh, my bad to skip the first part and just read "multiCluster"... sorry.